src/poetry/factory.py

```diff
@@ -125,31 +126,32 @@ def configure_sources(
             repository = cls.create_package_source(
                 source, config, disable_cache=disable_cache
             )
-            is_default = bool(source.get("default", False))
-            is_secondary = bool(source.get("secondary", False))
+            priority = Priority[source.get("priority", Priority.PRIMARY.name).upper()]
+            if source.get("default"):
```

Comment:
Does this need to display a warning so that users can migrate their existing `pyproject.toml`? We now only display warnings when calling `poetry source add`. `Source` also generates warnings, but `Factory` does not call that with legacy parameters.

Reply:
I have added a warning.